src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -931,7 +921,10 @@ QVariant QgsLegendModel::data( const QModelIndex &index, int role ) const
     {
       QgsExpressionContext expressionContext;
       if ( vlayer )
+      {
         connect( vlayer, &QgsVectorLayer::symbolFeatureCountMapChanged, this, &QgsLegendModel::forceRefresh );
```

Comment:
I am in favor of doing this connection and triggering the count next to each other. But won't we end up with dozens of connections if this connection is done inside the data method?

Reply:
Well since the data only acts at the layer level and this is only for layers with an expression, there is the possibility of having a few. 

Optimally this should occur only at the last layer containing an expression to evaluate but there is no reliable way to ensure this.

This means only one connection per layer to evaluate. Maybe people will go  crazy with this but from what I recall connection don't use that much resources, so I'm not too worried.

Otherwise the other solution that I had was too complex and finnicky with the tasks signals that kept track of queued counter in the task manager. This way is more elegant and will deal easily with the addition of expressions without much overhead and ensure that the legend is asynchronously updated.

Reply:
Connections itself don't use a lot of resources. Triggering expensive label re-evaluation various times upon count change does.
Can you at least add a `Qt::UniqueConnection` flag if there's no better place to put this call?

Reply:
Done